librz/bin/format/pe/pe.c

```diff
@@ -4048,6 +3876,120 @@ struct rz_bin_pe_lib_t *PE_(rz_bin_pe_get_libs)(struct PE_(rz_bin_pe_obj_t) * bi
 	return NULL;
 }
 
+RZ_OWN RzList /* RzBinSymbol */ *PE_(rz_bin_pe_get_clr_symbols)(struct PE_(rz_bin_pe_obj_t) * bin) {
+	if (!bin || !bin->clr || !bin->clr->methoddefs) {
+		return NULL;
+	}
+	RzList /* RzBinSymbol */ *methods = rz_list_newf((RzListFree)rz_bin_symbol_free);
+	if (!methods) {
+		return NULL;
+	}
+
+	// Namespace and classes
+
+	// Each typedef contains a methodlist field which indexes into
+	// the MethodDef table and marks the start of methods
+	// belonging to that type
+
+	// In order to determine the end of the methods of that type,
+	// we mark the start of the next run with `type_methods_end`
+	RzListIter *type_it = rz_list_iterator(bin->clr->typedefs);
+
+	char *type_name = NULL;
+	char *type_namespace = NULL;
+
+	ut32 type_methods_start = rz_list_length(bin->clr->methoddefs) + 1;
+	ut32 type_methods_end = type_methods_start;
+
+	if (type_it) {
+		Pe_image_metadata_typedef *typedef_ = type_it->data;
+		type_name = rz_buf_get_string(bin->clr->strings, typedef_->name);
+		type_namespace = rz_buf_get_string(bin->clr->strings, typedef_->namespace);
+
+		type_methods_start = ((Pe_image_metadata_typedef *)type_it->data)->methodlist;
+		type_methods_end = rz_list_length(bin->clr->methoddefs) + 1;
+
+		type_it = type_it->n;
+		if (type_it) {
+			type_methods_end = ((Pe_image_metadata_typedef *)type_it->data)->methodlist;
+		}
+	}
+
+	int i = 1;
+	RzListIter *it;
+	Pe_image_metadata_methoddef *methoddef;
+	rz_list_foreach (bin->clr->methoddefs, it, methoddef) {
+		if ((type_name || type_namespace) && i >= type_methods_start) {
```

Comment:
We usually reverse the condition and call `continue` then.

Reply:
I don't think this applies here? I want the loop body to run regardless of whether the condition holds.